frontend/src/component/common/MessageBanner/MessageBanner.tsx

```diff
@@ -0,0 +1,139 @@
+    justifyContent: 'center',
+    padding: theme.spacing(1),
+    gap: theme.spacing(1),
+    borderBottom: '1px solid',
+    borderColor: theme.palette[variant].border,
+    background: theme.palette[variant].light,
+    color: theme.palette[variant].dark,
+}));
+
+const StyledIcon = styled('div', {
+    shouldForwardProp: prop => prop !== 'variant',
+})<{ variant?: BannerVariant }>(({ theme, variant = 'neutral' }) => ({
+    display: 'flex',
+    alignItems: 'center',
+    color: theme.palette[variant].main,
+}));
+
+const StyledMessage = styled('div')(({ theme }) => ({
+    fontSize: theme.fontSizes.smallBody,
+}));
+
+const StyledLink = styled(Link)(({ theme }) => ({
+    fontSize: theme.fontSizes.smallBody,
+}));
+
+type BannerVariant = 'warning' | 'info' | 'error' | 'success' | 'neutral';
+
+interface IMessageFlag {
+    enabled: boolean;
+    message: string;
+    variant?: BannerVariant;
+    icon?: string;
+    link?: string;
+    linkText?: string;
+    plausibleEvent?: string;
+}
+
+// TODO: Grab a real feature flag instead
+const mockFlag: IMessageFlag = {
+    enabled: true,
+    message:
+        '<strong>Heads up!</strong> It seems like one of your client instances might be misbehaving.',
+    variant: 'warning',
+    link: '/admin/network',
+    linkText: 'View Network',
+    plausibleEvent: 'network_warning',
+};
+
+export const MessageBanner = () => {
+    const { uiConfig } = useUiConfig();
+
+    const { enabled, message, variant, icon, link, linkText, plausibleEvent } =
+        { ...mockFlag, enabled: uiConfig.flags.messageBanner };
+
+    if (!enabled) return null;
+
+    return (
+        <StyledBar variant={variant}>
+            <StyledIcon variant={variant}>
+                <BannerIcon icon={icon} variant={variant} />
+            </StyledIcon>
+            <StyledMessage dangerouslySetInnerHTML={{ __html: message }} />
+            <BannerButton
+                link={link}
+                linkText={linkText}
+                plausibleEvent={plausibleEvent}
+            />
+        </StyledBar>
+    );
+};
+
+interface IBannerIconProps {
+    icon?: string;
+    variant?: BannerVariant;
+}
+
+const BannerIcon = ({ icon, variant }: IBannerIconProps) => {
+    if (icon === 'none') return null;
+    if (icon) return <Icon>{icon}</Icon>;
+    if (variant) return <WarningAmber />;
+    // TODO: Add defaults for other variants?
+    return null;
+};
+
+interface IBannerButtonProps {
+    link?: string;
+    linkText?: string;
+    plausibleEvent?: string;
+}
+
+const BannerButton = ({
+    link,
+    linkText = 'More info',
+    plausibleEvent,
+}: IBannerButtonProps) => {
+    if (!link) return null;
+
+    const navigate = useNavigate();
+    const tracker = usePlausibleTracker();
+    const external = link.startsWith('http');
+
+    const trackEvent = () => {
+        if (!plausibleEvent) return;
+        tracker.trackEvent('message_banner', {
+            props: { event: plausibleEvent },
+        });
```

Comment:
I like the usage of `plausibleEvent` property allowing us to track different banners independently